### PR TITLE
Update to common cookie banner

### PR DESCRIPTION
**Pull request by kevsers**
#### What does this PR do? Please provide some context
This updates our current banner to the common cookie banner
#### Where should the reviewer start?
cookieBanner,js 
#### How can this be manually tested? (brief repro steps)
I will send the link to the deployment with these changes - here are some checks;
- Clean browser from cookies, upon launch of the site you should see the banner and only 2 essential cookies should be loaded
- Upon click on a link/button - the banner should be removed and non-essential cookies are allowed in
#### What are the relevant TFS items? (list id numbers)
101096
#### Definition of done:
- [x] Title of the pull request is clear and informative
- [ ] Add pull request hyperlink to relevant TFS items
- [ ] For large or complex change: schedule an in-person review session
- [ ] This change has appropriate test coverage

#### Reminders BEFORE merging
1. Get at least two approvals
1. If you're merging from a feature branch into the development branch then "flatten" or "squash" commits
1. If merging from the development branch into master (or porting changes from upstream) then use github's UI to get review feedback, but use the git command line interface to complete the actual merge.

#### Reminders AFTER merging
1. Delete the remote feature branch
1. Resolve relevant TFS items
1. (reverse merge) If you merged from the development branch into master then check to see if there are any changes in master that can be merged down to the development branch (like hotfixes, etc). In this case, use github's UI for feedback and the git command line interface for the actual merge.

[//]: # ( todo: If you merged into development branch then verify change in our "rolling deployment" environment. Then notify stakeholders interested in or involved with the change )

[//]: # ( fyi: This content was heavily inspired by )
[//]: # ( 1 Our team's policies and processes )
[//]: # ( 2 https://github.com/sprintly/sprint.ly-culture/blob/master/pr-template.md )
[//]: # ( 3 The book "The Checklist Manifesto: How to Get Things Right" by Atul Gawande )
[//]: # ( 4 https://github.com/Azure/azure-event-hubs/blob/master/.github/PULL_REQUEST_TEMPLATE.md )

### CODE REVIEW

**Review by manikarthikk — APPROVED** [review]

File: comprehensive/lms/static/cookieBanner.js [hunk 1]
```diff
@@ -53,33 +89,56 @@ oxa = window.oxa || {};
...
+$(document).ready(function() {
+    $.ajax({
+        url: "/cookies/get_cookies",
+        cache: true,
+        timeout: 30000,
+        success: function(data) {
+            cookieNotice.init(data);
+            $("#cookie-markup").html(cookieNotice.Markup);
+
+	    // by default hide the banner ux
```
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: please normalize indentation

Comment (sdolenc) [reply comprehensive/lms/static/cookieBanner.js]: yeah. this line contains tabs. i highly recommend using an editor like vs code

File: comprehensive/lms/static/cookieBanner.js [hunk 2]
```diff
@@ -53,33 +89,56 @@ oxa = window.oxa || {};
...
+            var css_link = document.createElement("link");
+            css_link.href = cookieNotice.Css;
+            css_link.rel = "stylesheet";
+            css_link.type = "text/css";
+            $("head").append(css_link);
+
+            // check if consent is still valid or we need to reconsent
+            var isConsentValid = false;
+            try {
+                if(cookieNotice.getCookie(cookieNotice.consentCookieName) > cookieNotice.MinimumConsentDate){
```
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: space after if

File: comprehensive/lms/static/cookieBanner.js [hunk 3]
```diff
@@ -53,33 +89,56 @@ oxa = window.oxa || {};
...
+                   isConsentValid = true;
+                }
+            } catch(Error){
```
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: space before {

File: comprehensive/lms/static/cookieBanner.js [hunk 4]
```diff
@@ -5,28 +5,64 @@ oxa = window.oxa || {};
...
 
-    // Click handler for the cookie consent
-    proto.cookieClickHandler = function () {
-        if (this.id !== "btnPrivacy") {
-            proto.closeCookieBanner();
-        }
+   // Adds function BI cookies
+    proto.addBICookies = function () {
+        // Allow BI cookies
+        (function (a, b, c, d) {
```
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: i'm blocking based on these variables names

File: comprehensive/lms/static/cookieBanner.js [hunk 5]
```diff
@@ -5,28 +5,64 @@ oxa = window.oxa || {};
...
+            d.src = a;
+            d.type = 'text/java' + c;
+            d.async = true;
+            a = b.getElementsByTagName(c)[0];
+            a.parentNode.insertBefore(d, a);
+        })();
     };
+   
+    // set the time that consent is given
+    proto.setConsentTime = function (){
```
Comment (sdolenc) [inline comprehensive/lms/static/cookieBanner.js]: nit space before {

**Review by sdolenc — CHANGES_REQUESTED** [review]
i did a cursory review. issues i found are minor and should be quick fixes

File: comprehensive/lms/static/cookieBanner.js (hunk 4 above)
Comment (kevsers) [reply comprehensive/lms/static/cookieBanner.js]: that script is given to us from the analytics team, I would prefer to keep them as is given

Comment (sdolenc) [reply comprehensive/lms/static/cookieBanner.js]: okay. i agree we shouldn't deviate in this case so that can more easily consume future updates.

**Review by sdolenc — APPROVED** [review]

**Review by ElakkiyaMohanRajan — APPROVED** [review]

**Pull request merged**